applications/ShallowWaterApplication/custom_utilities/shallow_water_utilities.cpp

```diff
@@ -100,109 +94,69 @@ void ShallowWaterUtilities::ComputeSmoothVelocity(ModelPart& rModelPart)
-        const double height = it_node->FastGetSolutionStepValue(HEIGHT);
-        const double velocity = norm_2(it_node->FastGetSolutionStepValue(VELOCITY));
-        it_node->FastGetSolutionStepValue(INTERNAL_ENERGY) = height + 0.5 * std::pow(velocity, 2);
-    }
-}
-
-void ShallowWaterUtilities::ComputeAccelerations(ModelPart& rModelPart)
-{
-    double dt_inv = rModelPart.GetProcessInfo()[DELTA_TIME];
-
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
-    {
-        auto it_node = rModelPart.NodesBegin() + i;
-
-        // Free suface derivative or vertical velocity
-        auto delta_surface = it_node->FastGetSolutionStepValue(FREE_SURFACE_ELEVATION) - it_node->FastGetSolutionStepValue(FREE_SURFACE_ELEVATION,1);
-        it_node->FastGetSolutionStepValue(VELOCITY_Z) = dt_inv * delta_surface;
-
-        // Acceleration
-        auto delta_vel = it_node->FastGetSolutionStepValue(VELOCITY) - it_node->FastGetSolutionStepValue(VELOCITY,1);
-        it_node->SetValue(ACCELERATION, dt_inv * delta_vel);
-    }
+    block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+        const double height = rNode.FastGetSolutionStepValue(HEIGHT);
+        const double velocity = norm_2(rNode.FastGetSolutionStepValue(VELOCITY));
+        rNode.FastGetSolutionStepValue(INTERNAL_ENERGY) = height + 0.5 * std::pow(velocity, 2);
+    });
 }
 
 void ShallowWaterUtilities::FlipScalarVariable(Variable<double>& rOriginVariable, Variable<double>& rDestinationVariable, ModelPart& rModelPart)
 {
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rModelPart.NumberOfNodes()); ++i)
-    {
-        auto it_node = rModelPart.NodesBegin() + i;
-        it_node->FastGetSolutionStepValue(rDestinationVariable) = -it_node->FastGetSolutionStepValue(rOriginVariable);
-    }
+    block_for_each(rModelPart.Nodes(), [&](NodeType& rNode){
+        rNode.FastGetSolutionStepValue(rDestinationVariable) = -rNode.FastGetSolutionStepValue(rOriginVariable);
+    });
 }
 
 void ShallowWaterUtilities::IdentifySolidBoundary(ModelPart& rSkinModelPart, double SeaWaterLevel, Flags SolidBoundaryFlag)
 {
-    #pragma omp parallel for
-    for (int i = 0; i < static_cast<int>(rSkinModelPart.NumberOfNodes()); ++i)
-    {
-        auto it_node = rSkinModelPart.NodesBegin() + i;
-        if (it_node->FastGetSolutionStepValue(TOPOGRAPHY) < SeaWaterLevel)
+    block_for_each(rSkinModelPart.Nodes(), [&](NodeType& rNode){
+        if (rNode.FastGetSolutionStepValue(TOPOGRAPHY) < SeaWaterLevel)
         {
-            it_node->Set(SolidBoundaryFlag, true);
+            rNode.Set(SolidBoundaryFlag, true);
         }
         else
         {
-            auto topography_gradient = it_node->GetValue(TOPOGRAPHY_GRADIENT);
-            auto normal = it_node->FastGetSolutionStepValue(NORMAL);
+            auto topography_gradient = rNode.GetValue(TOPOGRAPHY_GRADIENT);
+            auto normal = rNode.FastGetSolutionStepValue(NORMAL);
```

Comment:
```suggestion
            array_1d<double,3> normal = rNode.FastGetSolutionStepValue(NORMAL);
```
At some point we experienced some weird side effects when using `auto` for `array_1d<double,3>` types.